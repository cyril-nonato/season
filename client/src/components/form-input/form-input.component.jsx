import React from 'react'

import * as S from './form-input.styles';

const FormInput = ({name, type, label, onChange}) => {
  return ( 
    <S.Container>
      <S.Input data-testid='input' required name={name} onChange={onChange} type={type} placeholder={label} />
      <S.Label data-testid='label'>{label}</S.Label>
    </S.Container>
   );
}
 
export default FormInput;