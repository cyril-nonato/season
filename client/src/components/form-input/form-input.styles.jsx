import styled from 'styled-components';

export const Container = styled.div`
  margin: 1.5rem 0;
  width: 100%;
`

export const Input = styled.input`
  border: none;
  border-bottom: 2px solid #999;
  padding: .5rem 0;
  font-size: 1.7rem;
  display: inline-block;
  width: inherit;
  font-family: inherit;
  background-color: transparent;

  &:focus {
    outline: none;
  }

  &::placeholder {
    color: #ccc;
    text-transform: capitalize;
  }

  &:placeholder-shown ~ label {
    opacity: 0;
    visibility: hidden;
    transform: translateY(-3rem);
  }
`;

export const Label = styled.label`
  transform: translateY(1rem);
  transition: all .3s;
  display: block;
  font-size: 1.7rem;
  text-transform: capitalize;
`;