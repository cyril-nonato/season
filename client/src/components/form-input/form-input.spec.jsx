import React from 'react'
import {render, fireEvent, cleanup} from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'

import FormInput from './form-input.component'

import * as S from './form-input.styles'

afterEach(cleanup)

describe('Form Input Styled Test', () => {
  
  test('should match the snapshot', () => {
    const tree = renderer.create(<S.Container />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('margin', '1.5rem 0');
  });

  test('should match the snapshot', () => {
    const tree = renderer.create(<S.Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Form input test component', () => {
  const props = {
    name: 'name',
    type: 'text',
    label: 'some-label',
    onChange: jest.fn()
  }

  test('check for props', () => {
    const {getByTestId } = render(<FormInput {...props} />);
    const input = getByTestId('input');
    const label = getByTestId('label');
    fireEvent.change(input, {
      target: { value: 'some-test'}
    });
    expect(input).toHaveValue('some-test');
    expect(input).toHaveAttribute('name', props.name);
    expect(label).toHaveTextContent(props.label);
  });
});

