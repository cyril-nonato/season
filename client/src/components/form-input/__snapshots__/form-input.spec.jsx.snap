// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Input Styled Test should match the snapshot 1`] = `
.c0 {
  margin: 1.5rem 0;
  width: 100%;
}

<div
  className="c0"
/>
`;

exports[`Form Input Styled Test should match the snapshot 2`] = `
.c0 {
  border: none;
  border-bottom: 2px solid #999;
  padding: .5rem 0;
  font-size: 1.7rem;
  display: inline-block;
  width: inherit;
  font-family: inherit;
}

.c0:focus {
  outline: none;
}

.c0::-webkit-input-placeholder {
  color: #ccc;
  text-transform: capitalize;
}

.c0::-moz-placeholder {
  color: #ccc;
  text-transform: capitalize;
}

.c0:-ms-input-placeholder {
  color: #ccc;
  text-transform: capitalize;
}

.c0::placeholder {
  color: #ccc;
  text-transform: capitalize;
}

.c0:placeholder-shown ~ label {
  opacity: 0;
  visibility: hidden;
  -webkit-transform: translateY(-3rem);
  -ms-transform: translateY(-3rem);
  transform: translateY(-3rem);
}

<input
  className="c0"
/>
`;
